### PR TITLE
[JBJCA-1312] connection-url is required when dataSourceClass is provided

### PR DESCRIPTION
**Pull request by gaol**
Jira: https://issues.jboss.org/browse/JBJCA-1312

### CODE REVIEW

File: adapters/src/test/resources/h2-driver-datasource-ds.xml [hunk 1]
```diff
@@ -12,6 +12,7 @@
   </datasource>
   <drivers>
       <driver name="h2">
+         <driver-class>org.h2.Driver</driver-class>
```
Comment (gaol) [inline adapters/src/test/resources/h2-driver-datasource-ds.xml]: Provides both driver-class and datasource-class, but no connection_url, it should work.

**Comment by jesperpedersen** [comment]
We should do a similar check for data-source-class and connection properties.

We also need a pull request against the 1.2 branch for this

**Comment by gaol** [comment]
Is the `connectionProps.size() == 0` check enough for the connection properties? Because there is no way to know what is the `URL` property for `LocalManagedConnectionFactory` IMO. :-)

**Comment by jesperpedersen** [comment]
Yeah, I think so

**Pull request merged**